### ISSUE TITLE
Bug In Transparent Background

### ISSUE THREAD

**Issue by Shivansh-007** [kyazdani42/nvim-tree.lua#450]
On opening a new buffer, editor window after closing the previous one, the transparent background goes.

https://user-images.githubusercontent.com/69356296/121887168-0624ca80-cd34-11eb-9624-b9b14f90afd2.mov

**Comment by Shivansh-007** [comment]
I think it is somewhat related to #290.

**Comment by kyazdani42** [comment]
it looks like it is, a bit annoying actually, i'm really not sure how to fix that.

**Comment by francisco285** [comment]
@kyazdani42 It seems to be related to neovim/neovim/issues/14670

Steps to reproduce:
1. Open Neovim
2. `:NvimTreeToggle<CR>`
3. `<C-w>o`
4. `:new<CR>`

Try the steps above with current code, the issue is reproduced.
Now, replace [these lines](https://github.com/kyazdani42/nvim-tree.lua/blob/589c36e26f99486e7ab5e2e6920636ffd5f3ae2a/lua/nvim-tree/view.lua#L264-L266) with these:

```lua
function _G.Test()
  if not (vim.bo.ft == 'NvimTree') then return end
  for k, v in pairs(M.View.winopts) do
    local cmd
    if v == true then
      cmd = string.format('setlocal %s', k)
    elseif v == false then
      cmd = string.format('setlocal no%s', k, v)
    else
      cmd = string.format('setlocal %s=%s', k, v)
    end
    -- vim.wo[winnr][k] = v
    vim.cmd(cmd)
  end
end
vim.api.nvim_exec([[autocmd BufWinEnter,BufEnter * lua _G.Test()]], false)
```

... and the issue is not reproduced (of course the code above is very hacky, but it's just to give an idea)

**Comment by kyazdani42** [comment]
@francisco285 thanks a lot. I was sure there was something wrong with the neovim api when setting options through lua, it also broke some of my config options when i switched from `set ...` to `vim.o....`. I guess this is a bug in neovim and i don't know how long it will take to fix it, so i'm going to apply a patch to go back to setlocal once again !

**Comment by kyazdani42** [comment]
should be fixed in 4388338a6a1
